cli/cmd/root.go

```diff
@@ -175,20 +177,45 @@ func updateRootUsage() {
 	})
 }
 
-func wasEnvFlagProvided() bool {
+func getCmdShorthandFlags(cmd *cobra.Command) strset.Set {
+	shorthands := strset.New()
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		if flag.Shorthand != "" {
+			shorthands.Add(flag.Shorthand)
+		}
+	})
+	return shorthands
+}
+
+func wasEnvFlagProvided(cmd *cobra.Command) bool {
```

Comment:
I'm thinking of something like:
```golang
func wasEnvFlagProvided(cmd *cobra.Command) bool {
	envFlagProvided := false
	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
		if flag.Shorthand == "e" && flag.Changed {
			envFlagProvided = true
		}
	})

	return envFlagProvided
}
```

This should cover all scenarios: `-e <var>`, `-e=<var>`, `-e<var>`, `-we<var>`, `--env <var>`, `--env=<var>`. What do you think?

Reply:
Good catch, I will do it this way!